### PR TITLE
Add va-text-input USWDS v3 variation

### PR DESCRIPTION
**Pull request by jamigibbs**
## Chromatic

https://1266-text-input-us-wds-v3--60f9b557105290003b387cd5.chromatic.com/?path=/docs/uswds-va-text-input--default

## Description
Closes https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1266

## Testing done
Storybook

## Screenshots

Screenshot 2022-12-20 at 3 14 04 PM

Screenshot 2022-12-20 at 3 13 11 PM

Screenshot 2022-12-20 at 3 13 26 PM

Screenshot 2022-12-20 at 3 13 35 PM

## Tasks
- [x] Create a version of this component that imports and uses the USWDS CSS
- [x] Make any structural changes necessary and add a prop that determines which version of the component will be used
- [x] Check with Natalie to make sure the component and the design match
- [x] Add a USWDS story to Storybook
- [ ] Merge component
- [ ] Create a new release of component-library
- [ ] Update component-library dependency in vets-design-system-documentation to get the updated component-docs.json

## Definition of done
- [ ] USWDS version has been created and added to Storybook
- [ ] Design.va.gov has the latest version of component-library

### CODE REVIEW

**Comment by jamigibbs** [comment]
@jhonnyoddball I got this going for us but the one odd thing I'm seeing is that the input height is displaying much larger than the [v3 example](https://designsystem.digital.gov/components/text-input/).

This is what I'm expecting it to look like based on the [USWDS site](https://designsystem.digital.gov/components/text-input/) linked above:

Screen Shot 2022-12-08 at 3 22 38 PM

But in [Storybook](https://1266-text-input-us-wds-v3--60f9b557105290003b387cd5.chromatic.com/?path=/docs/uswds-va-text-input--default), the input is appearing larger than that:

Screen Shot 2022-12-08 at 3 23 14 PM

I'm not sure if I'm just missing a stylesheet import or if there's still something odd going on in the Storybook environment with Formation? Or is that actually now it's supposed to look? Something for us to investigate.

File: packages/web-components/src/components/va-text-input/va-text-input.scss [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  outline: none !important;
+}
+
+:host([uswds]) input:not([disabled]):focus{	
+  outline: 2px solid var(--color-gold-light);	
+  outline-offset: 2px;	
+}
+
+input {
+  box-sizing: border-box;
```
Comment (jamigibbs) [inline packages/web-components/src/components/va-text-input/va-text-input.scss:24]: This fixes the issue we were seeing with the input height calculating larger than what it should be.

https://github.com/department-of-veterans-affairs/component-library/pull/587#issuecomment-1343386102

**Comment by jamigibbs** [comment]
Hi @GnatalieH! Do you mind taking a look at this to make sure the component and the design match?

**Comment by GnatalieH** [comment]
Hi @jamigibbs. One thing I'm noticing is that there appears to be 4px extra padding between the label and the input field in the default, focus, and success states. The input field always has a top margin of 8px. So, there's usually 8px spacing between the bottom of the label and the top of the input field. When the error message line gets added in the error state, it includes 4px top and bottom padding. So I think if you can remove that extra 4px padding in the non-error states we'd be good. Thanks!

**Comment by jamigibbs** [comment]
@GnatalieH That's a good catch. The v3 `.usa-error-message` class was adding this space but I updated the structure so that the class doesn't appear unless there was an error message. Can you confirm that it's displaying as expected now?
 
https://1266-text-input-us-wds-v3--60f9b557105290003b387cd5.chromatic.com/?path=/docs/uswds-va-text-input--success

**Review by GnatalieH — APPROVED** [review]
This looks great, thanks! One question: does our label's font size look bigger than the USWDS version? When I check the CSS it seems the same between our version and theirs.

**Comment by jamigibbs** [comment]
I think it's because they are adding a font-family to the label class. It is loading "Helvetica Neue" in our Storybook but "Source Sans Pro" on the USWDS site:

```
.usa-label {
    font-family: "Source Sans Pro Web", "Helvetica Neue", Helvetica, Roboto, Arial, sans-serif;
    font-size: 1.06rem;
    line-height: 1.3;
    display: block;
    font-weight: normal;
    margin-top: 1.5rem;
    max-width: 30rem;
}
```

[USWDS Site](https://designsystem.digital.gov/components/text-input/)

Screenshot 2022-12-20 at 3 35 17 PM

[VA.gov Storybook](https://1266-text-input-us-wds-v3--60f9b557105290003b387cd5.chromatic.com/?path=/docs/uswds-va-text-input--default)

Screenshot 2022-12-20 at 3 34 54 PM

**Review by jhonnyoddball — APPROVED** [review]

**Pull request merged**